mmv1/third_party/terraform/services/apphub/resource_apphub_application_test.go

```diff
@@ -0,0 +1,226 @@
+	return enableApphubAPI(context) + acctest.Nprintf(`
+
+resource "google_apphub_application" "example2" {
+  location = "us-east1"
+  application_id = "tf-test-example-application%{random_suffix}"
+  display_name = "Application Full New%{random_suffix}"
+  scope {
+    type = "REGIONAL"
+  }
+  attributes {
+    environment {
+      type = "TEST"
+		}
+		criticality {  
+      type = "MISSION_CRITICAL"
+		}
+		business_owners {
+		  display_name =  "Alice%{random_suffix}"
+		  email        =  "alice@google.com%{random_suffix}"
+		}
+		developer_owners {
+		  display_name =  "Bob%{random_suffix}"
+		  email        =  "bob@google.com%{random_suffix}"
+		}
+		operator_owners {
+		  display_name =  "Charlie%{random_suffix}"
+		  email        =  "charlie@google.com%{random_suffix}"
+		}
+  }
+  depends_on = [google_project_service.apphub]
+}
+`, context)
+}
+
+func testAccApphubApplication_applicationUpdateCriticality(context map[string]interface{}) string {
+	return enableApphubAPI(context) + acctest.Nprintf(`
+
+resource "google_apphub_application" "example2" {
+  location = "us-east1"
+  application_id = "tf-test-example-application%{random_suffix}"
+  display_name = "Application Full New%{random_suffix}"
+  scope {
+    type = "REGIONAL"
+  }
+  attributes {
+    environment {
+      type = "TEST"
+		}
+		criticality {  
+      type = "MEDIUM"
+		}
+		business_owners {
+		  display_name =  "Alice%{random_suffix}"
+		  email        =  "alice@google.com%{random_suffix}"
+		}
+		developer_owners {
+		  display_name =  "Bob%{random_suffix}"
+		  email        =  "bob@google.com%{random_suffix}"
+		}
+		operator_owners {
+		  display_name =  "Charlie%{random_suffix}"
+		  email        =  "charlie@google.com%{random_suffix}"
+		}
+  }
+  depends_on = [google_project_service.apphub]
+}
+`, context)
+}
+
+func testAccApphubApplication_applicationUpdateOwners(context map[string]interface{}) string {
+	return enableApphubAPI(context) + acctest.Nprintf(`
+
+resource "google_apphub_application" "example2" {
+  location = "us-east1"
+  application_id = "tf-test-example-application%{random_suffix}"
+  display_name = "Application Full New%{random_suffix}"
+  scope {
+    type = "REGIONAL"
+  }
+  attributes {
+    environment {
+      type = "TEST"
+		}
+		criticality {  
+      type = "MEDIUM"
+		}
+		business_owners {
+		  display_name =  "Alice%{random_suffix}"
+		  email        =  "alice@google.com%{random_suffix}"
+		}
+		developer_owners {
+		  display_name =  "Bob%{random_suffix}"
+		  email        =  "bob@google.com%{random_suffix}"
+		}
+		developer_owners {
+			display_name =  "Derek%{random_suffix}"
+			email        =  "derek@google.com%{random_suffix}"
+		  }
+		operator_owners {
+		  display_name =  "Charlie%{random_suffix}"
+		  email        =  "charlie@google.com%{random_suffix}"
+		}
+  }
+  depends_on = [google_project_service.apphub]
+}
+`, context)
+}
+
+func enableApphubAPI(context map[string]interface{}) string {
```

Comment:
It doesn't look like we need a project for this, so it can be removed

Reply:
Done